File: Chapter5/exercises/factorial.py
# Import libraries
import time

# Factorial number
n = 10
# n = 20

# Start counter
start_time = time.time()

# Compute factorial
factorial = 1
for i in range(1, n + 1):
    factorial *= i

# Stop counter
end_time = time.time()

# Execution time
duration = end_time - start_time
print("Factorial result:", factorial)
print("Execution time:", duration, "seconds")

File: Chapter5/exercises/factorial.cpp
// Import libraries
#include <iostream>
#include <chrono>
using namespace std;
using namespace std::chrono;

// Main function
int main() {

    // Declare variables
    int n = 10;
    int n = 20;
    long factorial = 1;

    // Start counter
    high_resolution_clock::time_point start = high_resolution_clock::now();

    // Compute factorial
    for (int i = 1; i <= n; ++i) {
        factorial *= i;
    }

    // Stop counter
    high_resolution_clock::time_point stop = high_resolution_clock::now();

    // Execution time
    microseconds duration = duration_cast<microseconds>(stop - start);
    cout << "Factorial result: " << factorial << endl;
    cout << "Execution time: " << duration.count() / 1000000.0 << " seconds" << endl;

    return 0;

}